### PR TITLE
[jquery] `after()`, `append()`, `before()`, and `prepend()` can accept an array of JQuery

### LINKED ISSUE #20316: JQuery.append() raises compilation error when appending JQuery array
- [x] I tried using the `@types/jquery` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @leonard-thieu 

I have come across some code that I think should compile but does not and can simply replicate the issue with the following:

```
namespace Testing {
    export class TestClass {
        private _someElements: Array<JQuery>;
        private _someElement: JQuery;

        public Foo() {
            this._someElement.append(this._someElements);
        }
    }
}
```
This fails with the error TS2345 stating:

> Argument of type 'JQuery<HTMLElement>[]' is not assignable to parameter of type '(this: HTMLElement, index: number, html: string) => string | Element | Text | Comment | Node[] | ...'.
> Type 'JQuery<HTMLElement>[]' provides no match for the signature '(this: HTMLElement, index: number, html: string): string | Element | Text | Comment | Node[] | JQuery<Node>'.

### PR DESCRIPTION
**Pull request by leonard-thieu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- ~~Provide a URL to documentation or source code which provides context for the suggested changes:~~
- ~~Increase the version number in the header if appropriate.~~
- ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

Fixes #20316.

### CODE REVIEW

File: types/jquery/index.d.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ interface JQueryStatic<TElement extends Node = HTMLElement> {
      * @see {@link https://api.jquery.com/jQuery.Callbacks/}
      * @since 1.7
      */
-    Callbacks<T extends Function>(flags?: string): JQuery.Callbacks<T>;
+    Callbacks<T extends Function>(flags?: string): JQuery.Callbacks<Function>;
```
Comment (andy-ms) [inline types/jquery/index.d.ts]: The type parameter does nothing anymore.

**Comment by andy-ms** [comment]
@leonard-thieu Looks like there is a single remaining test failure in `jquery` itself.
```
ERROR: 67:9  expect  Expected type to be:
  JQuery.AjaxSettings
got:
  AjaxSettings<any>
```

**Comment by leonard-thieu** [comment]
Pushed new commit, not sure why PR isn't updating.

https://github.com/leonard-thieu/DefinitelyTyped/commits/jquery

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**